File: lib_js/models/structures/urlResult.js
export class UrlResult {
    constructor(data) {
        this.orig_url = data.orig_url || data.origUrl || null;
        this.resolved_url = data.resolved_url || data.resolvedUrl || null;
        this.title = data.title || null;
        this.source = data.source || null;
        this.description = data.description || null;
        this.image = data.image || null;
        this.image_data = data.image_data || data.imageData || null;
        this.publish_date = data.publish_date || data.publishDate || null;
        this.categories = data.categories || null;
        this.lang = data.lang || null;
        this.original_content =
            data.original_content || data.originalContent || null;
        this.metadata = data.metadata || null;
        this.human_readable_content =
            data.human_readable_content || data.humanReadableContent || null;
    }

    toString() {
        return `Original URL: ${this.orig_url}, Resolved URL: ${this.resolved_url}, Title: ${this.title}, Source: ${this.source}, Description: ${this.description}, Image: ${this.image}, Publish Date: ${this.publish_date}, Categories: ${this.categories}, Language: ${this.lang}`;
    }
}

File: source/models/structures/url_result.py
from typing import List, Optional
from pydantic import BaseModel
from datetime import datetime


class UrlResult(BaseModel):
    orig_url: Optional[str] = None
    resolved_url: Optional[str] = None
    title: Optional[str] = None
    source: Optional[str] = None
    description: Optional[str] = None
    image: Optional[str] = None
    image_data: Optional[list[dict]] = None
    publish_date: Optional[datetime] = None
    categories: Optional[List[str]] = None
    lang: Optional[str] = None
    # original_content: Optional[str] = None
    metadata: Optional[str] = None
    human_readable_content: Optional[str] = None

    def __str__(self):
        return f"Original URL: {self.orig_url}, Resolved URL: {self.resolved_url}, Title: {self.title}, Source: {self.source}, Description: {self.description}, Image: {self.image}, Publish Date: {self.publish_date}, Categories: {self.categories}, Language: {self.lang}"
